components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -392,7 +393,7 @@ void OutputManager::run(const util::TimeStamp& timestamp)
       snapshot_start = m_case_t0;
       snapshot_start += m_time_bnds[0];
     }
-    if (not filespecs.storage.snapshot_fits(snapshot_start)) {
+    if (filespecs.is_open and not filespecs.storage.snapshot_fits(snapshot_start)) {
```

Comment:
Unopened and not full cannot happen. We never close a file until full (in this if statement), and a non-full file after a restart is opened at init time, so that we _never_ have unopened not full files.

Reply:
Yes. When performing a restart, the last output file (if not full) is opened during init. Precisely b/c when we then call run, we are in the same situation as we were during the run (i.e., with a semi-full file still open).

Reply:
From rhist (this PR fixes the issue you reported).